nifi-framework-bundle/nifi-framework/nifi-client-dto/src/main/java/org/apache/nifi/web/api/dto/status/ProcessGroupStatusSnapshotDTO.java

```diff
@@ -563,28 +563,66 @@ public ProcessGroupStatusSnapshotDTO clone() {
+            final List<ConnectionStatusSnapshotEntity> collectionStatusSnapshotEntities = new ArrayList<>();
+            for (final ConnectionStatusSnapshotEntity connectionStatusSnapshotEntity :  getConnectionStatusSnapshots()) {
+                ConnectionStatusSnapshotEntity clone = connectionStatusSnapshotEntity.clone();
+                clone.setId(connectionStatusSnapshotEntity.getId());
+                collectionStatusSnapshotEntities.add(clone);
+            }
+            other.setConnectionStatusSnapshots(collectionStatusSnapshotEntities);
+        }
+
+        if (getProcessorStatusSnapshots() != null) {
+            final List<ProcessorStatusSnapshotEntity> processorStatusSnapshotEntities = new ArrayList<>();
+            for (final ProcessorStatusSnapshotEntity processorStatusSnapshotEntity :  getProcessorStatusSnapshots() ) {
+                ProcessorStatusSnapshotEntity clone = processorStatusSnapshotEntity.clone();
+                clone.setId(processorStatusSnapshotEntity.getId());
+                processorStatusSnapshotEntities.add(clone);
+            }
+            other.setProcessorStatusSnapshots(processorStatusSnapshotEntities);
+        }
+
+        if (getRemoteProcessGroupStatusSnapshots() != null) {
+            final List<RemoteProcessGroupStatusSnapshotEntity> remoteProcessGroupStatusSnapshotEntities = new ArrayList<>();
+            for (final RemoteProcessGroupStatusSnapshotEntity remoteProcessGroupStatusSnapshotEntity :  getRemoteProcessGroupStatusSnapshots() ) {
+                RemoteProcessGroupStatusSnapshotEntity clone = remoteProcessGroupStatusSnapshotEntity.clone();
+                clone.setId(remoteProcessGroupStatusSnapshotEntity.getId());
+                remoteProcessGroupStatusSnapshotEntities.add(clone);
+            }
+            other.setRemoteProcessGroupStatusSnapshots(remoteProcessGroupStatusSnapshotEntities);
+        }
+
+        if (getInputPortStatusSnapshots() != null) {
+            final List<PortStatusSnapshotEntity> portStatusSnapshotEntities = new ArrayList<>();
+            for (final PortStatusSnapshotEntity portStatusSnapshotEntity :  getInputPortStatusSnapshots()) {
+               PortStatusSnapshotEntity clone = portStatusSnapshotEntity.clone();
+               clone.setId(portStatusSnapshotEntity.getId());
+               portStatusSnapshotEntities.add(clone);
+            }
+            other.setInputPortStatusSnapshots(portStatusSnapshotEntities);
+        }
+
+        if (getOutputPortStatusSnapshots() != null) {
+            final List<PortStatusSnapshotEntity> portStatusSnapshotEntities = new ArrayList<>();
+            for (final PortStatusSnapshotEntity portStatusSnapshotEntity :  getOutputPortStatusSnapshots()) {
+                PortStatusSnapshotEntity clone = portStatusSnapshotEntity.clone();
+                clone.setId(portStatusSnapshotEntity.getId());
+                portStatusSnapshotEntities.add(clone);
+            }
```

Comment:
Minor optimization, it looks like the Port Status Snapshot creation could be use a shared method to create a copy.